src/impl/load.c

```diff
@@ -0,0 +1,289 @@
+#include "load.h"
+
+#include "../ent/game_history.h"
+#include "../util/io_util.h"
+#include "../util/string_util.h"
+#include "config.h"
+#include "gcg.h"
+#include <ctype.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <regex.h>
+
+char *get_xt_gcg_string(const char *identifier, ErrorStack *error_stack) {
+  char *game_id_str = NULL;
+
+  // Check if this is a Cross-tables URL
+  if (strstr(identifier, "cross-tables.com") != NULL) {
+      // Extract game ID from URL parameter u=XXXXX using regex
+      regex_t regex;
+      regmatch_t matches[2];
+      
+      int regex_status = regcomp(&regex, "u=([0-9]+)", REG_EXTENDED);
+      if (regex_status != 0) {
```

Comment:
since the regex is constant failing compilation would be pretty bad, so we should do log_fatal instead I think.